alacritty/src/config/mod.rs

```diff
@@ -112,10 +115,21 @@ pub fn load(options: &Options) -> Config {
 
     // Override config with CLI options.
     options.override_config(&mut config);
+    // Override the values that have limits.
+    override_with_limits(&mut config);
 
     config
 }
 
+// Some config values have limits. We allow users to give values that go beyond the limit,
+// but override them here.
+fn override_with_limits(config: &mut Config) {
+    config.ui_config.window.dimensions = window::Dimensions::new(
+        Column(max(MIN_COLS, config.ui_config.window.dimensions.columns_u32() as usize)),
+        Line(max(MIN_LINES, config.ui_config.window.dimensions.lines_u32() as usize)),
+    )
+}
```

Comment:
This isn't really a great way of doing things, which is why this function hasn't existed before. Why are you doing this while loading the config, rather than just relying on the checks we have to do anyways?

It might be useful to warn! when loading the config with values that are too small, but it's not really super important given that basically the only size that doesn't work is 1x1 and 0 cols/lines, at that point we can basically do whatever we want anyways. You could use NonZeroXXX datatypes for deserialization then the only case would be 1x1. In theory with a custom deserializer you can even warn in all cases, but I don't really see how it is necessary.

Reply:
Was also thinking about this myself... Let me remove this change.